noahvarghese/capstone-server
# Fixed tests, dev, prod, and coverage

- some small tweaks and added in new routes, adjusted permissions again
- displays roles and departments barely
- Added fields to return
- added field and hopefully fixed tests
- fixed errors and formatted files
- pluralized routes
- fixed typo
- tweaked the types for test data, added some functionality to test helpler, added test cases
- added tests pass to create role, department and users
- added feature files for deleting records
- added checks to determine what tests to run
- added checks to ci
- added another badge
- formatting issues
- refactored
- added documentation checks
- should work this time
- debug ci
- working on checking routes
- execute shell script
- got script for documentation checking working
- fixed documentation stage
- final update to routes
- fixed comments
- trying composite actions
- try using name
- Added back in ci, and changed error codes
- moved items over to custom action
- removed continue on error
- forgot file extension
- test update
- testing
- test
- log return value
- testing with nested action call
- fixed typo
- changed format
- change multiple files for test
- parse comma sooner
- trying to pass output without relying on error codes
- fixed type coercion
- didnt actually save
- didn't pass it properly
- improper use of aquare brackets
- did not realize all output gets saved
- fixed exiting
- try again
- forgot to close quotes
- final modifications to new actions, some tidy up amongst scripts
- fixed up routes so that documentation shouldnt error anymore
- added in resetting of database for environment if there were changes
- abstracted out some more bash
- added extra output
- fixed double step
- limit api tests run in ci
- added other files to check before running tests
- renamed tests
- rerun tests if tests change
- changed step name
- removed actions i made to run tests, using jest's built in helper now, and pulled ci out into composite action
- forgot to checkout before using
- testing my if statment
- pre changed output
- try again
- check location
- checkout twice, hopefully works
- pass in changed
- check for git dit
- formatting
- more tests
- try again with fetch
- see if this works
- change fetch depth
- removed some debug outputs
- hopefully fixed variable interpolation
- try with double quotes
- removed secondary checkout
- moved items back into own action
- forgot to fetch before hand
- fetch twice because we need a deep fetch the for the action and I wont always remember
- shallow fetch first
- reworking tests to only use jest
- added in some changes to jest to help with turnaround time, added in reporting of coverage and test status
- typo in script
- updated paths to badges
- fixed paths
- Added missing dependency
- trying differenct hashes
- added in ci for badges
- removed from gitignore
- removed from gitignore
- removed from gitignore
- configure storing artifacts, caching dependencies
- added back newlines
- absolute paths
- relative paths
- fixed typo
- improperly types command
- closed brackets, removed shell
- fixed shell
- check cache
- typo
- no square brackets
- single squares
- changed cache dirs
- run install no matter what
- moved all test utilities to support jest instead of cucumber, fixed models, just need to finish porting api tests
- removed old file
- broke dev script
- fixed build step
- all tests pass again, finalized migration from cucumber to jest
- enhanced error messages and functionality
- fixed
- tried to refactor the test setups but failed transferring items between test anv and the tests
- forgot to remove reference to deleted file
- removed unused types
- removed build output in source
- reordered tsconfig options
- yay got dev script working again
- added forced test result output
- test successful
- fixed the routes test, fixed code coverage
